src/core/jit/exec.rs

```diff
@@ -118,7 +117,7 @@ where
                     if let Some(array) = value.as_array() {
                         join_all(field.nested_iter().map(|field| {
                             join_all(array.iter().enumerate().map(|(index, value)| {
-                                let ctx = ctx.with_value(value); // Output::JsonArray::Value
+                                let ctx = ctx.with_value(value).with_field(field); // Output::JsonArray::Value
```

Comment:
seems like we always setting both field and value. Would it be better to merge it into single operation?